mycli/main.py

```diff
@@ -1014,7 +1016,16 @@ def cli(database, user, host, port, socket, password, dbname,
                   defaults_file=defaults_file, login_path=login_path,
                   auto_vertical_output=auto_vertical_output, warn=warn,
                   myclirc=myclirc)
-
+    if list_dsn :
+        try:
+            for alias in mycli.config['alias_dsn']:
+                click.secho(alias + " : " + mycli.config['alias_dsn'][alias])
+            sys.exit(0)
+        except Exception as err:
```

Comment:
Could you please change `Exception` to the exception you're trying to catch? Now every exception in the try block will be reported as `Invalid DSN`. This `except Exception as e` is used in other places but there the exception is reported without translation (`click.secho(str(e), err=True, fg='red')`).

Reply:
exact, i add keyerror